packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerHighlight/utils/textBoxMapping/CellProvider.ts

```diff
@@ -0,0 +1,96 @@
+import { isSideBySideOnLine } from '../common/bboxUtils';
+import { TextLayoutCell, TextLayoutCellBase } from '../textLayout/types';
+
+export class CellProvider {
+  private readonly skippedCells: TextLayoutCellBase[] = [];
+  private cells: TextLayoutCellBase[]; // make sure to handle this as immutable array
```

Comment:
Then why not use `Object.freeze()` to actually make it immutable?

Reply:
:+1: I was't aware that I can use the type annotation `readonly TextLayoutCellBase[];`